### PR TITLE
Export scalar 'Alpha' values.

### LINKED ISSUE #784: baseColorFactor is always set to 1.0 no matter what the materials alpha says
**Describe the bug**

baseColorFactor is always set to 1.0
```
             "baseColorFactor" : [
                    0.800000011920929,
                    0.800000011920929,
                    0.800000011920929,
                    1.0
                ],
```

It should be set to whatever the material specifies:

image

**To Reproduce**
Create a box, set its material to be just a color. Set the alpha to some number, export, notice how the alpha was set to 1

**Version**
 - Blender Version 2.81

### PR DESCRIPTION
**Pull request by donmccurdy**
Fixes #784.

### CODE REVIEW

File: addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_materials_pbr_metallic_roughness.py [hunk 1]
```diff
@@ -47,6 +47,9 @@ def __filter_pbr_material(blender_material, export_settings):
 
 
 def __gather_base_color_factor(blender_material, export_settings):
+    alpha_socket = gltf2_blender_get.get_socket(blender_material, "Alpha")
+    alpha = alpha_socket.default_value if alpha_socket is not None and not alpha_socket.is_linked else 1
```
Comment (scurest) [inline addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_materials_pbr_metallic_roughness.py:51]: `1` -> `1.0` to fix the tests.

**Comment by scurest** [comment]
Can you apply this too? It removes the importer code that handles the exporter not understanding the alpha socket.

<details>

<summary>Patch</summary>

```diff
diff --git a/addons/io_scene_gltf2/blender/imp/gltf2_blender_pbrMetallicRoughness.py b/addons/io_scene_gltf2/blender/imp/gltf2_blender_pbrMetallicRoughness.py
index 00bd08d..deb9e30 100644
--- a/addons/io_scene_gltf2/blender/imp/gltf2_blender_pbrMetallicRoughness.py
+++ b/addons/io_scene_gltf2/blender/imp/gltf2_blender_pbrMetallicRoughness.py
@@ -233,7 +233,7 @@ def base_color(
         base_color_factor = [1, 1, 1, 1]

     if base_color_texture is None and not mh.vertex_color:
-        color_socket.default_value = base_color_factor
+        color_socket.default_value = base_color_factor[:3] + [1]
         if alpha_socket is not None:
             alpha_socket.default_value = base_color_factor[3]
         return
@@ -242,10 +242,7 @@ def base_color(
     needs_color_factor = base_color_factor[:3] != [1, 1, 1]
     needs_alpha_factor = base_color_factor[3] != 1.0 and alpha_socket is not None
     if needs_color_factor or needs_alpha_factor:
-        # For now, always create the color factor node because the exporter
-        # reads the alpha value from here. Can get rid of "or needs_alpha_factor"
-        # when it learns to understand the alpha socket.
-        if needs_color_factor or needs_alpha_factor:
+        if needs_color_factor:
             node = mh.node_tree.nodes.new('ShaderNodeMixRGB')
             node.label = 'Color Factor'
             node.location = x - 140, y
@@ -255,7 +252,7 @@ def base_color(
             # Inputs
             node.inputs['Fac'].default_value = 1.0
             color_socket = node.inputs['Color1']
-            node.inputs['Color2'].default_value = base_color_factor
+            node.inputs['Color2'].default_value = base_color_factor[:3] + [1]

         if needs_alpha_factor:
             node = mh.node_tree.nodes.new('ShaderNodeMath')
```

</details>

File: addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_materials_pbr_metallic_roughness.py (hunk 1 above)
Comment (donmccurdy) [reply addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_materials_pbr_metallic_roughness.py]: Thanks!

**Comment by donmccurdy** [comment]
Great, done! :)

**Pull request merged**